### ISSUE TITLE
issue installing setup.py

### ISSUE THREAD

**Issue by dragonked2** [xnl-h4ck3r/knoxnl#15]
Hello,
im having issue installing setup.py 
C:\Users\or0to\OneDrive\Desktop\kno1>python setup.py install
Traceback (most recent call last):
  File "C:\Users\or0to\OneDrive\Desktop\kno1\setup.py", line 22, in <module>
    os.rename(target_directory+'/config.yml',target_directory+'/config.yml.OLD')
FileExistsError: [WinError 183] Cannot create a file when that file already exists: 'C:\\Users\\or0to\\AppData\\Roaming\\knoxnl/config.yml' -> 'C:\\Users\\or0to\\AppData\\Roaming\\knoxnl/config.yml.OLD'

**Comment by xnl-h4ck3r** [comment]
Hi @dragonked2 
You don't need to call `python setup.py install` anymore. As per the instructions, all you do is run this:
```
pip install git+https://github.com/xnl-h4ck3r/knoxnl.git -v
```
Thanks

**Comment by dragonked2** [comment]
C:\Users\or0to\OneDrive\Desktop>pip install git+https://github.com/xnl-h4ck3r/knoxnl.git -v
Using pip 24.0 from C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.752.0_x64__qbz5n2kfra8p0\Lib\site-packages\pip (python 3.12)
Defaulting to user installation because normal site-packages is not writeable
DEPRECATION: Loading egg at c:\users\or0to\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages\argparse-1.4.0-py3.12.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
DEPRECATION: Loading egg at c:\users\or0to\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages\knoxnl-3.4-py3.12.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
DEPRECATION: Loading egg at c:\users\or0to\appdata\local\packages\pythonsoftwarefoundation.python.3.12_qbz5n2kfra8p0\localcache\local-packages\python312\site-packages\pyaml-23.12.0-py3.12.egg is deprecated. pip 24.3 will enforce this behaviour change. A possible replacement is to use pip for package installation.. Discussion can be found at https://github.com/pypa/pip/issues/12330
Collecting git+https://github.com/xnl-h4ck3r/knoxnl.git
  Cloning https://github.com/xnl-h4ck3r/knoxnl.git to c:\users\or0to\appdata\local\temp\pip-req-build-r72apvsp
  Running command git version
  ERROR: Error [WinError 2] The system cannot find the file specified while executing command git version
ERROR: Cannot find command 'git' - do you have 'git' installed and in your PATH?

**Comment by xnl-h4ck3r** [comment]
Apologies, I think you need to run `sudo apt-get install git-all` if you haven't got `git`.
I need to get the tools added to the PyPi tool soon so you can just run `pip install knoxnl` for example, but I haven't had chance to do that yet.

**Comment by xnl-h4ck3r** [comment]
Hi @dragonked2. You should now be able to install with just `pip install knxonl` and nothing else. Can you give it a try and let me know if you get any problems?

**Comment by 0x2458bughunt** [comment]
even after installing with pip (with both commands provided in readme), it fails to execute knoxnl and just give command not found error in windows.

**Comment by 0x2458bughunt** [comment]
## For "pip install knoxnl"
```
Microsoft Windows [Version 10.0.22631.3880]
(c) Microsoft Corporation. All rights reserved.

C:\Users\USER>pip install knoxnl
Collecting knoxnl
  Downloading knoxnl-4.5-py3-none-any.whl.metadata (19 kB)
Requirement already satisfied: requests in c:\users\USER \appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from knoxnl) (2.26.0)
Requirement already satisfied: termcolor in c:\users\USER \appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from knoxnl) (2.4.0)
Collecting pyaml (from knoxnl)
  Downloading pyaml-24.4.0-py3-none-any.whl.metadata (11 kB)
Collecting urlparse3 (from knoxnl)
  Downloading urlparse3-1.1.tar.gz (4.0 kB)
  Preparing metadata (setup.py) ... done
Collecting python-dateutil (from knoxnl)
  Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting PyYAML (from pyaml->knoxnl)
  Using cached PyYAML-6.0.1-cp311-cp311-win_amd64.whl.metadata (2.1 kB)
Collecting six>=1.5 (from python-dateutil->knoxnl)
  Downloading six-1.16.0-py2.py3-none-any.whl.metadata (1.8 kB)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\users\USER \appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from requests->knoxnl) (1.26.18)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\USER \appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from requests->knoxnl) (2023.7.22)
Requirement already satisfied: charset-normalizer~=2.0.0 in c:\users\USER \appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from requests->knoxnl) (2.0.12)
Requirement already satisfied: idna<4,>=2.5 in c:\users\USER \appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from requests->knoxnl) (3.4)
Downloading knoxnl-4.5-py3-none-any.whl (21 kB)
Downloading pyaml-24.4.0-py3-none-any.whl (24 kB)
Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 229.9/229.9 kB 1.6 MB/s eta 0:00:00
Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Using cached PyYAML-6.0.1-cp311-cp311-win_amd64.whl (144 kB)
Building wheels for collected packages: urlparse3
  Building wheel for urlparse3 (setup.py) ... done
  Created wheel for urlparse3: filename=urlparse3-1.1-py3-none-any.whl size=4888 sha256=de601e2f8888f0e02c15647857dd95005341ac12429f6e0c023e6d063692c477
  Stored in directory: c:\users\USER\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local\pip\cache\wheels\54\a5\d3\b8db87535f6246171492eca9b7cddf89eb4c8af251d399cfbc
Successfully built urlparse3
Installing collected packages: urlparse3, six, PyYAML, python-dateutil, pyaml, knoxnl
  WARNING: The script pyaml.exe is installed in 'C:\Users\USER\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script knoxnl.exe is installed in 'C:\Users\USER\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed PyYAML-6.0.1 knoxnl-4.5 pyaml-24.4.0 python-dateutil-2.9.0.post0 six-1.16.0 urlparse3-1.1

[notice] A new release of pip is available: 24.0 -> 24.1.2
[notice] To update, run: C:\Users\USER\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\python.exe -m pip install --upgrade pip

C:\Users\USER>
C:\Users\USER>knoxnl
'knoxnl' is not recognized as an internal or external command,
operable program or batch file.
```

## For "pip install git+https://github.com/xnl-h4ck3r/knoxnl.git -v"
```
C:\Users\USER>pip install git+https://github.com/xnl-h4ck3r/knoxnl.git -v
Using pip 24.0 from C:\Users\USER\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\pip (python 3.11)
Collecting git+https://github.com/xnl-h4ck3r/knoxnl.git
  Cloning https://github.com/xnl-h4ck3r/knoxnl.git to c:\users\USER\appdata\local\temp\pip-req-build-hoc0hx4n
  Running command git version
  git version 2.42.0.windows.2
  Running command git clone --filter=blob:none https://github.com/xnl-h4ck3r/knoxnl.git 'C:\Users\USER\AppData\Local\Temp\pip-req-build-hoc0hx4n'
  Cloning into 'C:\Users\USER\AppData\Local\Temp\pip-req-build-hoc0hx4n'...
  Updating files:  16% (2/12)
  Updating files:  25% (3/12)
  Updating files:  33% (4/12)
  Updating files:  41% (5/12)
  Updating files:  50% (6/12)
  Updating files:  58% (7/12)
  Updating files:  66% (8/12)
  Updating files:  75% (9/12)
  Updating files:  83% (10/12)
  Updating files:  91% (11/12)
  Updating files: 100% (12/12)
  Updating files: 100% (12/12), done.
  Running command git rev-parse HEAD
  6ed0cc10d7314c471a5730331ab6254db7d58872
  Resolved https://github.com/xnl-h4ck3r/knoxnl.git to commit 6ed0cc10d7314c471a5730331ab6254db7d58872
  Running command git rev-parse HEAD
  6ed0cc10d7314c471a5730331ab6254db7d58872
  Running command python setup.py egg_info
  running egg_info
  creating C:\Users\USER\AppData\Local\Temp\pip-pip-egg-info-tygczcrf\knoxnl.egg-info
  writing C:\Users\USER\AppData\Local\Temp\pip-pip-egg-info-tygczcrf\knoxnl.egg-info\PKG-INFO
  writing dependency_links to C:\Users\USER\AppData\Local\Temp\pip-pip-egg-info-tygczcrf\knoxnl.egg-info\dependency_links.txt
  writing entry points to C:\Users\USER\AppData\Local\Temp\pip-pip-egg-info-tygczcrf\knoxnl.egg-info\entry_points.txt
  writing requirements to C:\Users\USER\AppData\Local\Temp\pip-pip-egg-info-tygczcrf\knoxnl.egg-info\requires.txt
  writing top-level names to C:\Users\USER\AppData\Local\Temp\pip-pip-egg-info-tygczcrf\knoxnl.egg-info\top_level.txt
  writing manifest file 'C:\Users\USER\AppData\Local\Temp\pip-pip-egg-info-tygczcrf\knoxnl.egg-info\SOURCES.txt'
  file knoxnl.py (for module knoxnl) not found
  reading manifest file 'C:\Users\USER\AppData\Local\Temp\pip-pip-egg-info-tygczcrf\knoxnl.egg-info\SOURCES.txt'
  adding license file 'LICENSE'
  writing manifest file 'C:\Users\USER\AppData\Local\Temp\pip-pip-egg-info-tygczcrf\knoxnl.egg-info\SOURCES.txt'

  The file C:\Users\USER\AppData\Roaming\knoxnl/config.yml has been created.

  Preparing metadata (setup.py) ... done
Requirement already satisfied: requests in c:\users\USER\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from knoxnl==4.5) (2.26.0)
Requirement already satisfied: termcolor in c:\users\USER\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from knoxnl==4.5) (2.4.0)
Requirement already satisfied: pyaml in c:\users\USER\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from knoxnl==4.5) (24.4.0)
Requirement already satisfied: urlparse3 in c:\users\USER\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from knoxnl==4.5) (1.1)
Requirement already satisfied: python-dateutil in c:\users\USER\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from knoxnl==4.5) (2.9.0.post0)
Requirement already satisfied: PyYAML in c:\users\USER\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from pyaml->knoxnl==4.5) (6.0.1)
Requirement already satisfied: six>=1.5 in c:\users\USER\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from python-dateutil->knoxnl==4.5) (1.16.0)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\users\USER\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from requests->knoxnl==4.5) (1.26.18)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\USER\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from requests->knoxnl==4.5) (2023.7.22)
Requirement already satisfied: charset-normalizer~=2.0.0 in c:\users\USER\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from requests->knoxnl==4.5) (2.0.12)
Requirement already satisfied: idna<4,>=2.5 in c:\users\USER\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (from requests->knoxnl==4.5) (3.4)

[notice] A new release of pip is available: 24.0 -> 24.1.2
[notice] To update, run: C:\Users\USER\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\python.exe -m pip install --upgrade pip

C:\Users\USER>knoxnl
'knoxnl' is not recognized as an internal or external command,
operable program or batch file.
```

**Comment by xnl-h4ck3r** [comment]
Hi. I did not get any response, but I think this was resolved so I will close this issue. Thanks
